cmd/probe/eventhandler.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/cmd/controller"
+	libudevwrapper "github.com/openebs/node-disk-manager/pkg/udev"
+)
+
+type EventAction string
+
+const (
+	AttachEA EventAction = libudevwrapper.UDEV_ACTION_ADD    // AttachEA is atach disk event name
+	DetachEA EventAction = libudevwrapper.UDEV_ACTION_REMOVE // DetachEA is detach disk event name
+)
+
+// ProbeEvent staruct contain a copy of controller it will update disk resources
+type ProbeEvent struct {
+	Controller *controller.Controller
+}
+
+// EventMessage struct contains attribute of event message info.
+type EventMessage struct {
+	Action  string                 // Action is event action like attach/detach
+	Devices []*controller.DiskInfo // list of disks details
+}
+
+// addDiskEvent fill disk details from different probes and push it to etcd
+func (pe *ProbeEvent) addDiskEvent(msg EventMessage) {
+	for _, diskDetails := range msg.Devices {
+		func() {
+			glog.Info("processing data for ", diskDetails.ProbeIdentifiers.Uuid)
+			diskDetails.HostName = pe.Controller.HostName
+			diskDetails.Uuid = diskDetails.ProbeIdentifiers.Uuid
```

Comment:
As per the diskDetails -- this should be an generated uuid by the ndm. Is there some change here?

Reply:
If we want to change disk uid generation functionality then we need this. Earlier @pawanpraka1  told we may revisit this uid generation function. As main disk discover and attach/detach event is generated by udev probe only at the time of discovering or event it generate one uid and that uid is same as disk uid that's why both are same.